### ISSUE TITLE
AnyOf not supported

### ISSUE THREAD

**Issue by JbIPS** [fastify/fast-json-stringify#49]
When trying to validate against a JSON schema containing `anyOf` type, the parser throw an error.
```json
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "properties": {
    "key1": {
      "anyOf": [
        {
          "type": "string"
        },
        {
          "type": "object"
        }
      ]
    }
  },
  "additionalProperties": false
}
```
throws `undefined unsupported`.

I'll make a PR with a failing test case so it can be clearer.

**Comment by JbIPS** [comment]
I looked a bit at the code and I'm thinking of adding a `case "anyOf"` that will call the function again for each key. 

What do you think?

**Comment by mcollina** [comment]
Can you add a reference to the part of the JSON Schema for this?

👍 for a PR anyway.

**Comment by JbIPS** [comment]
> Can you add a reference to the part of the JSON Schema for this?

I'm sorry, do you mean the [JSON Schema spec](http://json-schema.org/latest/json-schema-validation.html#rfc.section.6.27) ?

**Comment by mcollina** [comment]
Yes thanks!! a PR would be very welcome!

**Comment by JbIPS** [comment]
I created #50 (and started it with failing tests) but I'm not sure how to start the implementation. My first guess was to add a case "anyOf" that will call the function again for each key.

What do you think of that?

**Comment by mcollina** [comment]
go for it!

**Comment by JbIPS** [comment]
Worked a bit on that yesterday and I'm a bit stuck. A achieve to catch `anyOf` properties and iterate over the children but I can't see how I could tell to 'try' each one until one pass. I though the method `$as[...]` would throw errors if the type doesn't match but they don't.

Any leads?

Thanks

**Comment by mcollina** [comment]
I don’t know really. Trying multiple paths and throwing errors completely defeat the purpose of this library.

A possibility might be to use Ajv to validate all the various option, but it would be slower.

**Comment by JbIPS** [comment]
If it's noted clearly that using `anyOf` cannot lead to any optimization but is supported, it could worth it, don't you think?

**Comment by mcollina** [comment]
Yes definitely.

**Comment by JbIPS** [comment]
I finally had some time to work on this but I'm hitting an issue: how can I use `ajv` to validate the data *inside* the generated code? I tried to require it but `require` is not defined :/

**Comment by mcollina** [comment]
It should be done following this pattern: https://github.com/fastify/fast-json-stringify/blob/master/index.js#L90-L92

**Comment by JbIPS** [comment]
Thank you! First draft of my implementation ready [here](https://github.com/fastify/fast-json-stringify/pull/50)

**Comment by JbIPS** [comment]
Closed by #50
